assets/src/edit-story/components/library/panes/media/mediaPane.js

```diff
@@ -182,69 +222,97 @@ function MediaPane(props) {
     )
     .map((attachment) => getResourceFromAttachment(attachment));
 
+  const refContainer = useRef();
+  const refContainerFooter = useRef();
+
+  useIntersectionEffect(
+    refContainerFooter,
+    (entry) => {
+      if (!isMediaLoaded || isMediaLoading) return;
+      if (!hasMore) return;
+      if (!entry.isIntersecting) return;
+
+      setPage({ page: page + 1 });
```

Comment:
Looking more into the use here, I don't see why we need both `page` and `setPage`. I think if we instead introduce `loadNextPage()` method in our context, it'd solve all of these nits: we won't be leaking bulk of the API and the meaning of this will be very clear.

Reply:
Agree, but I would call it `setNextPage` because it doesn't load the next page by itself.